cmd/ocm-backplane/cloud/common.go

```diff
@@ -0,0 +1,128 @@
+package cloud
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/backplane-cli/pkg/awsutil"
+	"github.com/openshift/backplane-cli/pkg/utils"
+	"io"
+)
+
+const OldFlowSupportRole = "role/RH-Technical-Support-Access"
+
+var StsClientWithProxy = awsutil.StsClientWithProxy
+var AssumeRoleWithJWT = awsutil.AssumeRoleWithJWT
+var NewStaticCredentialsProvider = credentials.NewStaticCredentialsProvider
+var AssumeRoleSequence = awsutil.AssumeRoleSequence
+
+type assumeChainResponse struct {
+	AssumptionSequence []namedRoleArn `json:"assumptionSequence"`
+}
+
+type namedRoleArn struct {
+	Name string `json:"name"`
+	Arn  string `json:"arn"`
+}
+
+func getIsolatedCredential(clusterID string) (aws.Credentials, error) {
+	if clusterID == "" {
+		return aws.Credentials{}, errors.New("must provide non-empty cluster ID")
+	}
+
+	ocmToken, err := utils.DefaultOCMInterface.GetOCMAccessToken()
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("failed to retrieve OCM token: %w", err)
+	}
+
+	email, err := utils.GetStringFieldFromJWT(*ocmToken, "email")
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("unable to extract email from given token: %w", err)
+	}
+
+	bpConfig, err := GetBackplaneConfiguration()
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("error retrieving backplane configuration: %w", err)
+	}
+
+	if bpConfig.AssumeInitialArn == "" {
+		return aws.Credentials{}, errors.New("backplane config is missing required `assume-initial-arn` property")
+	}
+
+	initialClient, err := StsClientWithProxy(bpConfig.ProxyURL)
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("failed to create sts client: %w", err)
+	}
+
+	seedCredentials, err := AssumeRoleWithJWT(*ocmToken, bpConfig.AssumeInitialArn, initialClient)
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("failed to assume role using JWT: %w", err)
+	}
+
+	backplaneClient, err := utils.DefaultClientUtils.MakeRawBackplaneAPIClientWithAccessToken(bpConfig.URL, *ocmToken)
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("failed to create backplane client with access token: %w", err)
+	}
+
+	response, err := backplaneClient.GetAssumeRoleSequence(context.TODO(), clusterID)
+	if err != nil {
+		return aws.Credentials{}, fmt.Errorf("failed to fetch arn sequence: %w", err)
+	}
+	if response.StatusCode != 200 {
```

Comment:
Let's use HTTP status code from the package instead.

```suggestion
	if response.StatusCode != http.StatusOK {
```